extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/radio/FrontierSiliconRadioApiResult.java

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.fsinternetradio.internal.radio;
+
+import java.io.IOException;
+import java.io.StringReader;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.CharacterData;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+/**
+ * This class hold the result of a request read from the radio. Upon a request the radio returns a XML document like
+ * this:
+ *
+ * <pre>
+ * <xmp>
+ * 	<fsapiResponse> <status>FS_OK</status> <value><u8>1</u8></value> </fsapiResponse>
+ * </xmp>
+ * </pre>
+ *
+ * This class parses this XML data and provides functions for reading and casting typical fields.
+ *
+ * @author Rainer Ostendorf
+ * @author Patrick Koenemann
+ *
+ */
+public class FrontierSiliconRadioApiResult {
+
+    /**
+     * XML structure holding the parsed response
+     */
+    final Document xmlDoc;
+
+    private static final Logger logger = LoggerFactory.getLogger(FrontierSiliconRadioApiResult.class);
+
+    /**
+     * Create result object from XML that was received from the radio.
+     *
+     * @param requestResultString
+     *            The XML string received from the radio.
+     */
+    public FrontierSiliconRadioApiResult(String requestResultString) {
+        Document xml = null;
+        try {
+            xml = getXmlDocFromString(requestResultString);
+        } catch (Exception e) {
+            logger.error("converting to XML failed: '" + requestResultString + "' with " + e.getClass().getName() + ": "
```

Comment:
this should only be debugging or trace level - and the message prints out the full XML, which is potentially huge. So you might log the e.getMessage() as debug, but the requestResultString should definitely only be trace level.